matlab/src/matlab/+arrow/+array/TimestampArray.m

```diff
@@ -30,15 +30,15 @@
         end
 
         function dates = toMATLAB(obj)
-            time = obj.Proxy.toMATLAB();
+            epochTime = obj.Proxy.toMATLAB();
 
-            epoch = datetime(1970, 1, 1, TimeZone="UTC");
+            timeZone = obj.Type.TimeZone;
+            ticksPerSecond = obj.Type.TimeUnit.ticksPerSecond();
 
-            tz = obj.Type.TimeZone;
-            ticsPerSecond = ticksPerSecond(obj.Type.TimeUnit);
-            
-            dates = datetime(time, ConvertFrom="epochtime", Epoch=epoch, ...
-                TimeZone=tz, TicksPerSecond=ticsPerSecond);
+            % UNIX Epoch (January 1st, 1970).
+            unixEpoch = datetime(0, ConvertFrom="posixtime");
```

Comment:
I think you need to specify `TimeZone="UTC"` here for the conversion to work.

Reply:
Thanks for catching this!

Reply:
Fixed.